crates/turborepo-lockfiles/src/bun/mod.rs

```diff
@@ -68,10 +147,137 @@ impl Lockfile for BunLockfile {
     }
 
     fn turbo_version(&self) -> Option<String> {
-        unimplemented!()
+        let (_, entry) = self.package_entry("turbo")?;
+        Some(entry.version().to_owned())
     }
 
     fn human_name(&self, package: &crate::Package) -> Option<String> {
-        unimplemented!()
+        let entry = self.packages.get(&package.key)?;
+        Some(entry.key.clone())
+    }
+}
+
+impl BunLockfile {
+    pub fn from_bytes(input: &[u8]) -> Result<Self, super::Error> {
+        let s = std::str::from_utf8(input).map_err(Error::from)?;
+        Self::from_str(s)
+    }
+
+    // Given a specific key for a package, return the most specific key that is
+    // present in the lockfile
+    fn package_entry(&self, key: &str) -> Option<(&str, &PackageEntry)> {
+        let (key, entry) =
+            PossibleKeyIter::new(key).find_map(|k| self.packages.get_key_value(k))?;
+        Some((key, entry))
+    }
+}
+
+impl FromStr for BunLockfile {
+    type Err = super::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parsed_json = biome_json_parser::parse_json(
+            s,
+            JsonParserOptions::default().with_allow_trailing_commas(),
+        );
+        if parsed_json.has_errors() {
+            // TODO
+        }
+        let syntax_tree = parsed_json.syntax();
+        let format = biome_json_formatter::format_node(
```

Comment:
if you want a slightly shorter trip, I think you can use `biome_deserialize` to get a `serde_json::Value`, then use serde to deserialize that into `BunLockfile`.

Reply:
I'm not seeing [`serde_json::Value`](https://docs.rs/biome_deserialize/latest/biome_deserialize/trait.Deserializable.html#foreign-impls) as being supported by `biome_deserialize`.